pkg/virt-launcher/virtwrap/network/binding_test.go

```diff
@@ -0,0 +1,165 @@
+package network
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/network/cache/fake"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("BindingMechanism", func() {
+	var (
+		TestRunPlug = func(driver BindMechanism, podInterfaceName string) {
+			err := driver.discoverPodNetworkInterface(podInterfaceName)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(driver.preparePodNetworkInterface()).To(Succeed())
+			Expect(driver.decorateConfig(driver.generateDomainIfaceSpec())).To(Succeed())
+		}
+	)
+	Context("when slirp binding mechanism is selected", func() {
+		var (
+			domain                      *api.Domain
+			driver                      SlirpBindMechanism
+			vmi                         *kubevirtv1.VirtualMachineInstance
+			NewDomainWithSlirpInterface = func() *api.Domain {
+				domain := &api.Domain{}
+				domain.Spec.Devices.Interfaces = []api.Interface{{
+					Model: &api.Model{
+						Type: "e1000",
+					},
+					Type:  "user",
+					Alias: api.NewUserDefinedAlias("default"),
+				},
+				}
+
+				// Create network interface
+				if domain.Spec.QEMUCmd == nil {
+					domain.Spec.QEMUCmd = &api.Commandline{}
+				}
+
+				if domain.Spec.QEMUCmd.QEMUArg == nil {
+					domain.Spec.QEMUCmd.QEMUArg = make([]api.Arg, 0)
+				}
+
+				return domain
+			}
+			newVMISlirpInterface = func(namespace string, name string) *kubevirtv1.VirtualMachineInstance {
+				vmi := newVMI(namespace, name)
+				vmi.Spec.Domain.Devices.Interfaces = []kubevirtv1.Interface{*kubevirtv1.DefaultSlirpNetworkInterface()}
+				kubevirtv1.SetObjectDefaults_VirtualMachineInstance(vmi)
+				return vmi
+			}
+		)
+		BeforeEach(func() {
+			domain = NewDomainWithSlirpInterface()
+			vmi = newVMISlirpInterface("testnamespace", "testVmName")
+			api.NewDefaulter(runtime.GOARCH).SetObjectDefaults_Domain(domain)
+			driver = SlirpBindMechanism{iface: &vmi.Spec.Domain.Devices.Interfaces[0], domain: domain}
+		})
+		It("Should create an interface in the qemu command line and remove it from the interfaces", func() {
+			TestRunPlug(&driver, "")
```

Comment:
I'm OK with this as is, but we could make it a bit more readable by either:

1. pass a variable
```golang
podIfaceName := ""
TestRunPlug(&driver, podIfaceName)
```

2.  have a specific "plug" function for slirp, which doesn't use the pod interface name, thus making it explicit slirp is not entirely normal
```golang
TestRunSlirpPlug() := func(driver BindMechanism) {
    podIfaceName := ""
    TestRunPlug(&driver, podIfaceName)
...

TestRunPlugSlirp(&driver)
...
}
```

Reply:
I am ok but will call this `TestRunPlugWithoutIfaceName`

Reply:
Done